packages/server/lib/cloud/protocol.ts

```diff
@@ -320,7 +322,11 @@ export class ProtocolManager implements ProtocolManagerShape {
     debug(`uploading %s to %s with a file size of %s`, filePath, uploadUrl, fileSize)
 
     try {
-      await putProtocolArtifact(filePath, dbSizeLimit(), uploadUrl)
+      const samplingInterval = process.env.CYPRESS_TEST_REPLAY_UPLOAD_SAMPLING_INTERVAL ?
```

Comment:
What is a user passes 'foo' to this env var, it'd evaluate to NaN, then what?

Reply:
Addressed in https://github.com/cypress-io/cypress/pull/30235/commits/dda7577511ba9a983bceafe2ccb0b1c7e2cb7d24